pkg/clusters/addons/kong/builder.go

```diff
@@ -36,6 +40,28 @@ func (b *Builder) WithPostgreSQL() *Builder {
 	return b
 }
 
+// WithEnterprise deploying kong enterpise
+func (b *Builder) WithEnterprise() *Builder {
+	b.enterprise = true
+	b.repo = DefaultEnterpriseImageRepo
+	b.tag = DefaultEnterpriseImageTag
+	b.license = KongLicenseSecretName
+	return b
+}
+
+// WithImage specify docker image repo and tag
+func (b *Builder) WithImage(repo, tag string) *Builder {
+	b.repo = repo
+	b.tag = tag
+	return b
+}
+
+// WithLicense specify license secret name
```

Comment:
Why take the secret name of the license instead of the license contents? We already have the potential in this builder to generate the secret, if we take the contents we can automate the secret generation (and cleanup).

I think I understand that the intention here is to allow someone to provide an already provisioned license secret, however given that we create the namespace for this addon deployment and we would want store I think it's just as well to manage the secret automatically and let the caller provide the actual secret contents.

Reply:
KTF is using helm charts deploy image, the secret name is required ,according to the docs https://github.com/Kong/charts/blob/main/charts/kong/values.yaml#L705-L709.

Reply:
The helm chart needs the secret name yes, but this library provides automation (the helm chart is used as an implementation detail of that automation) it should be just as well to completely automate the license management and just accept the license itself, which is the bare minimum we need to perform that automation.

Reply:
@shaneutt  We did license management including deployment (https://github.com/Kong/kubernetes-testing-framework/pull/104/files#diff-3d68f23e86db256aa769ae3f812e898e238f059fd95d96ff32a452981f588070R177) and cleanup (https://github.com/Kong/kubernetes-testing-framework/pull/104/files#diff-3d68f23e86db256aa769ae3f812e898e238f059fd95d96ff32a452981f588070R228). 
Can I create a follow up issue for your requirement ?

Reply:
>  Can I create a follow up issue for your requirement ?

I may be OK with doing some of this in a follow up, however can you help me to better understand how the end user is meant to provide the _actual license contents_? It was appearing that the license secret name was provided via `WithLicense()`, but that the license contents was provided via an environment variable. Rather than have two strongly different ways to provide configuration to configure the addon, I would like to see us at least implement a configuration method to provide the actual license data _in addition_ to the option to change the secret name. Does that sound like an OK compromise?

Reply:
@shaneutt I think generally it is not suggested to provide actual license content through function interface/ or shell scrip/ or CLI is a security concern
Yes, The code only apply user license (the secret) if it is specified using interface https://github.com/Kong/kubernetes-testing-framework/pull/104/files#diff-7cc1dbd395c3c0ea6e62a7d5a3d4493f37054d7b7be23f198340cdd6a52c0be2R60.

Reply:
https://github.com/Kong/kubernetes-testing-framework/pull/104/files#diff-3d68f23e86db256aa769ae3f812e898e238f059fd95d96ff32a452981f588070R178